cc/sgx/services/fns/sgx_trusted_lib/src/trusted_worker/operators.rs

```diff
@@ -0,0 +1,280 @@
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// Insert std prelude in the top for the sgx feature
+use std::convert::TryInto;
+use std::iter::FromIterator;
+#[cfg(feature = "mesalock_sgx")]
+use std::prelude::v1::*;
+use std::vec;
+
+use crate::eigen_crypto::sign::ecdsa::KeyPair;
+use crate::trusted_worker::register_func;
+use crate::worker::{Worker, WorkerContext};
+use eigen_core::{Error, ErrorKind, Result};
+
+use num_bigint::{BigInt, ToBigInt};
+use num_bigint::{BigUint, ToBigUint};
+
+pub struct OperatorWorker {
+    worker_id: u32,
+    func_name: String,
+    input: Option<OperatorWorkerInput>,
+}
+
+impl OperatorWorker {
+    pub fn new() -> Self {
+        OperatorWorker {
+            worker_id: 0,
+            func_name: "operator".to_string(),
+            input: None,
+        }
+    }
+}
+
+enum OperatorKind {
+    AddCipherCipher,
+    AddCipherPlain,
+    SubCipherCipher,
+    SubCipherPlain,
+    Encrypt,
+    Decrypt,
+}
+
+struct OperatorWorkerInput {
+    // TODO: Use a more property field name to describe what we want to save here.
+    //       e.g., when operator is `Encrypt`, `operand_1` is not a cipher one
+    op: OperatorKind,
+    operand_1: String,
+    operand_2: String,
+}
+
+impl Worker for OperatorWorker {
+    fn function_name(&self) -> &str {
+        self.func_name.as_str()
+    }
+
+    fn set_id(&mut self, worker_id: u32) {
+        self.worker_id = worker_id;
+    }
+
+    fn id(&self) -> u32 {
+        self.worker_id
+    }
+
+    fn prepare_input(&mut self, dynamic_input: Option<String>) -> Result<()> {
+        let msg = dynamic_input.ok_or_else(|| Error::from(ErrorKind::InvalidInputError))?;
+
+        // `args` should be "op,op_num,op1,op2,op3,..."
+        // number is parsed with big endian
+
+        let splited = msg.split(",").collect::<Vec<_>>();
+
+        if splited.len() <= 2 {
+            return Err(Error::from(ErrorKind::InvalidInputError));
+        }
+
+        let op = splited[0];
+
+        match op {
+            "add_cipher_cipher" | "add_cipher_plain" | "sub_cipher_cipher" | "sub_cipher_plain" => {
+                let op_num = splited[1]
+                    .parse::<u8>()
+                    .map_err(|_| Error::from(ErrorKind::InvalidInputError))?;
+
+                if op_num != 2 && splited.len() != 4 {
+                    return Err(Error::from(ErrorKind::InvalidInputError));
+                }
+
+                let operand_1 = splited[2];
+                let operand_2 = splited[3];
+
+                let op_kind = match op {
+                    "add_cipher_cipher" => OperatorKind::AddCipherCipher,
+                    "add_cipher_plain" => OperatorKind::AddCipherPlain,
+                    "sub_cipher_cipher" => OperatorKind::SubCipherCipher,
+                    "sub_cipher_plain" => OperatorKind::SubCipherPlain,
+                    _ => unreachable!(),
```

Comment:
Will this lead to fns crash?

Reply:
I think  InvalidInputError  is more friendly as a return when invalid op_kind be passed.